File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        available = collections.Counter(magazine)

        for char in ransomNote:
            if char not in available:
                return False

            elif available[char] == 1:
                del available[char]
            
            else:
                available[char] -= 1

        return True
        

File: 0383-ransom-note/0383-ransom-note.go
func canConstruct(ransomNote string, magazine string) bool {
    
    count := make(map[rune]int)
    
    for _, letter := range magazine {
        count[letter]++
    }

    for _, letter := range ransomNote {
        if _, ok := count[letter]; !ok {
            return false
        } else if count[letter] == 1{
            delete(count, letter)
        } else {
            count[letter]--
        }
    }
    return true
}